docs/upgrading-to-v0.18.x.md

```diff
@@ -0,0 +1,244 @@
+successful upgrade by still following all migration steps skipping the 6th one
+(i.e. upgrade your provider package).
+
+## Migration Steps
+
+1. Stop the Crossplane controllers by scaling down Crossplane deployment to 0
+   replicas. This is to ensure that our manual steps throughout this document
+   not to get overridden by any existing composite resource composing resources
+   with the objects that we are interacting here.
+
+   ```
+   kubectl -n crossplane-system scale deployment crossplane --replicas=0
+   ```
+
+2. Export and store your `NodePool` and `GKECluster` resources in a local file
+   as a reference for importing after upgrade.
+
+   ```
+   kubectl get nodepools.container.gcp.crossplane.io -o yaml > my-nodepools.yaml
+   kubectl get gkeclusters.container.gcp.crossplane.io -o yaml > my-gkeclusters.yaml
+   ```
+
+   Verify the contents of the files, before moving further.
+
+
+4. Mark your `NodePool` and `GKECluster` resources for `deletionPolicy` `Orphan`
+   to make sure that they are not deleted once custom resource objects in the
+   cluster deleted.
+
+   ```
+   kubectl get nodepools.container.gcp.crossplane.io -o name | xargs -n1 kubectl patch  -p '{"spec": {"deletionPolicy": "Orphan"}}' --type merge
+   kubectl get gkeclusters.container.gcp.crossplane.io -o name | xargs -n1 kubectl patch  -p '{"spec": {"deletionPolicy": "Orphan"}}' --type merge
+   ```
+
+5. Delete orphaned objects of type `NodePool` and `GKECluster`.
+
+   ```
+   kubectl delete nodepools.container.gcp.crossplane.io --all
+   kubectl delete gkeclusters.container.gcp.crossplane.io --all
+   ```
+   
+6. Delete the deprecated CRD types.
+   ```
+   kubectl delete crd nodepools.container.gcp.crossplane.io
+   kubectl delete crd gkeclusters.container.gcp.crossplane.io
+   ```
+
+
+7. Upgrade your provider package to `v0.18.x` by changing `spec.package` in 
+   `provider.pkg.crossplane.io` resource.
+
+
+8. Start the Crossplane controllers by scaling Crossplane deployment back
+   to 1 replicas.
+
+   ```
+   kubectl -n crossplane-system scale deployment crossplane --replicas=1
+   ```
+
+9. Wait until provider package to be `INSTALLED` and `HEALTHY`.
+
+   ```
+   kubectl get provider.pkg.crossplane.io
+   ```
+
+### Importing existing NodePools and Clusters
+
+At this point, you would need to [import] your existing `NodePool` and
+`GKECluster` (which renamed as `Cluster`) resources using the new schema.
+See the API changes for each resource in the description of the
+[the PR switching v1].
+
+Now, you can either use `NodePool` and `Cluster` types in [provider-gcp] or 
+[provider-gcp-beta] depending on your needs (i.e. relying on a beta feature or
+not).
+
+#### Importing as Stable
+
+To import your resources for stable types, use the following templates by
+filling required fields using the information from `my-nodepools.yaml` and
+`my-gkeclusters.yaml` files that we created in the beginning of migration steps.
+
+**Template to import `Cluster`:**
```

Comment:
@turkenh could these import templates just be a find and replace of GVK on the exported CRDs?

Reply:
@hasheddan could you elaborate a bit? 

Do you mean reusing the same templates for beta and stable by just replacing GVK?

Reply:
@turkenh i was actually thinking of literally using the internet exported files, and just doing a find and replace on them

Reply:
But what about the schema changes? Thinking about sandboxType in Nodepool for example, would we also instruct to rename it as per new schema?

Or dp you mean doing this for beta only since there would be no schema change?